packages/siopv2-oid4vp-rp-rest-client/src/agent/SiopV2OID4VpRpRestClient.ts

```diff
@@ -0,0 +1,96 @@
+import { fetch } from 'cross-fetch'
+import {
+  ISiopV2OID4VpRpRestClient,
+  IGenerateAuthRequestArgs,
+  IGenerateAuthRequestURIResponse,
+  IGetAuthStatusArgs,
+  IRequiredContext,
+  IRemoveAuthRequestSessionArgs,
+} from '../index'
+import Debug from 'debug'
+import { IAgentPlugin } from '@veramo/core'
+
+const debug = Debug('ssi-sdk-siopv2-oid4vp-rp-rest-client:SiopV2OID4VpRpRestClient')
+// todo merge this with Niels's branch feature/siop-verifier and use those classes/definitions
+export class SiopV2OID4VpRpRestClient implements IAgentPlugin {
+  readonly methods: ISiopV2OID4VpRpRestClient = {
+    removeAuthRequestSession: this.removeAuthRequestSession.bind(this),
+    generateAuthRequest: this.generateAuthRequest.bind(this),
+    getAuthStatus: this.getAuthStatus.bind(this),
+  }
+
+  private readonly baseUrl?: string
+  private readonly definitionId?: string
+
+  constructor(baseUrl?: string, definitionId?: string) {
+    if (baseUrl) {
+      this.baseUrl = baseUrl
+    }
+    if (definitionId) {
+      this.definitionId = definitionId
+    }
+  }
+
+  private async removeAuthRequestSession(args: IRemoveAuthRequestSessionArgs, context: IRequiredContext): Promise<void> {
+    const baseUrl = this.checkBaseUrlParameter(args.baseUrl)
+    const definitionId = this.checkDefinitionIdParameter(args.definitionId)
+    fetch(this.uriWithBase(`/webapp/definitions/${definitionId}/auth-requests/${args.correlationId}`, baseUrl), {
+      method: 'DELETE',
+    })
+  }
+
+  private async getAuthStatus(args: IGetAuthStatusArgs, context: IRequiredContext): Promise<any> {
+    const baseUrl = this.checkBaseUrlParameter(args.baseUrl)
+    const url = this.uriWithBase('/webapp/auth-status', baseUrl)
+    const definitionId = this.checkDefinitionIdParameter(args.definitionId)
+    const statueResponse = await fetch(url, {
+      method: 'POST',
+      headers: {
+        'Content-Type': 'application/json',
+      },
+      body: JSON.stringify({
+        correlationId: args.correlationId,
+        definitionId,
+      }),
+    })
+    debug(`auth status response: ${statueResponse}`)
+    const success = statueResponse && statueResponse.status >= 200 && statueResponse.status < 400
+    if (success) {
+      return await statueResponse.json()
+    }
+    throw Error(`Statue has returned ${statueResponse.status}`)
```

Comment:
First check whether the REST endpoint in my branch returns an error object or not. If that is the case return that in the error.

 also a typo in statusResponse

Reply:
yes you're returning it/ changed to get the response json, if any error occurs , I throw a general error here